src/PowerupSubsystem.cpp

```diff
@@ -88,10 +101,19 @@ const bool PowerupSubsystem::is_powerup(const int object_id) const {
 }
 
 glm::vec3 PowerupSubsystem::get_next_powerup_position() const {
-    float x = (rand() % (2 * (int) POWERUP_LOCATION_LIMITS.x)) - (int) POWERUP_LOCATION_LIMITS.x;
-    float y = 1.5f;
-    float z = (rand() % (2 * (int) POWERUP_LOCATION_LIMITS.z)) - (int) POWERUP_LOCATION_LIMITS.z;
-    return glm::vec3(x, y, z);
+
+    if (power_loc >= 20) {
+        power_loc = 0;
+    }
+
+    glm::vec3 location = charcoal_locations[power_loc];
```

Comment:
The z location needs to be modified so it is based on how high the mound is. Right now the packets are in the highest mounds, and way above the lowest mounds.